metaflow/plugins/argo/argo_workflows.py

```diff
@@ -360,6 +449,8 @@ def _compile(self):
                 )
                 # Set the entrypoint to flow name
                 .entrypoint(self.flow.name)
+                # Set succeeded lifecycle hook
+                .lifecycle_hooks(self.flow.name, ignore_events=self._ignore_events)
```

Comment:
nit - we can skip setting the lifecycle hook if `ignore_events` is set to True rather than embedding the skip logic inside the method.